ocs_ci/deployment/deployment.py

```diff
@@ -469,24 +478,40 @@ def deploy_ocs_via_operator(self, image=None):
         self.subscribe_ocs()
         operator_selector = get_selector_for_ocs_operator()
         subscription_plan_approval = config.DEPLOYMENT.get("subscription_plan_approval")
-        package_manifest = PackageManifest(
-            resource_name=defaults.OCS_OPERATOR_NAME,
-            selector=operator_selector,
-            subscription_plan_approval=subscription_plan_approval,
-        )
-        package_manifest.wait_for_resource(timeout=300)
+        ocs_version = config.ENV_DATA["ocs_version"]
+        if Version.coerce(ocs_version) >= Version.coerce("4.9"):
+            ocs_operator_names = [
+                defaults.ODF_OPERATOR_NAME,
+                defaults.OCS_OPERATOR_NAME,
+            ]
+        else:
+            ocs_operator_names = [defaults.OCS_OPERATOR_NAME]
         channel = config.DEPLOYMENT.get("ocs_csv_channel")
-        csv_name = package_manifest.get_current_csv(channel=channel)
-        csv = CSV(resource_name=csv_name, namespace=self.namespace)
-        if (
-            config.ENV_DATA["platform"] == constants.IBMCLOUD_PLATFORM
-            and not live_deployment
-        ):
-            csv.wait_for_phase("Installing", timeout=720)
-            logger.info("Sleeping for 30 seconds before applying SA")
-            time.sleep(30)
-            link_all_sa_and_secret_and_delete_pods(constants.OCS_SECRET, self.namespace)
-        csv.wait_for_phase("Succeeded", timeout=720)
+        is_ibm_sa_lined = False
+        for ocs_operator_name in ocs_operator_names:
+            package_manifest = PackageManifest(
+                resource_name=ocs_operator_name,
+                selector=operator_selector,
+                subscription_plan_approval=subscription_plan_approval,
+            )
+            package_manifest.wait_for_resource(timeout=300)
+            csv_name = package_manifest.get_current_csv(channel=channel)
+            csv = CSV(resource_name=csv_name, namespace=self.namespace)
+            if (
+                config.ENV_DATA["platform"] == constants.IBMCLOUD_PLATFORM
+                and not live_deployment
+            ):
+                csv.wait_for_phase("Installing", timeout=720)
+                if not is_ibm_sa_lined:
+                    logger.info("Sleeping for 30 seconds before applying SA")
+                    time.sleep(30)
+                    link_all_sa_and_secret_and_delete_pods(
+                        constants.OCS_SECRET, self.namespace
+                    )
+                    is_ibm_sa_lined = True
+                    logger.info("Deleting all pods in openshift-storage namespace")
+                    exec_cmd(f"oc delete pod --all -n {self.namespace}")
```

Comment:
Not sure, if I understand all the internals correctly, but what would happen, once this will be executed for the second operator? Wouldn't it be already in _Succeeded_ state and thus it will timeout on the line 504?

Reply:
I am running now some verification also for IBM cloud deployment, just to make sure:
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/5723/console